app/0.14.x/cli.md

```diff
@@ -213,4 +213,158 @@ Options:
+     exit (this will drop the remaining active connections).
+
+[Back to TOC](#table-of-contents)
+
+---
+
+### Controlled stop using the Kong CLI
+
+When controlling Kong through the CLI, make sure you have stopped traffic from
+being routed to the Kong node to tear down (steps 1 and 2 above).
+
+Now to actually stop the Kong node steps 3 to 5 can be executed at once by the
+`kong quit` command. Assuming a 10-second wait for connections to close,
+execute:
+
+```
+kong quit --timeout=10
+```
+
+The output will be either `Kong stopped (gracefully)` or
+`Timeout, Kong stopped forcefully`, depending on whether the timeout was hit.
+
+[Back to TOC](#table-of-contents)
+
+---
+
+### Controlled stop using Docker commands
+
+Similar to working with the CLI, make sure you have stopped traffic from
+being routed to the Kong container to tear down (steps 1 and 2 above).
+
+When using `docker stop` to stop a container, Docker will by default use the
+`SIGTERM` signal, and an addition `SIGKILL` after a timeout. Since this will drop
+active connections when used with Kong we have
+to use a different way to send `SIGQUIT` (step 3). This can be done with the
+`docker kill` command:
+
+```
+docker kill ----signal=SIGQUIT [container id]
+```
+
+Now, if after the timeout (step 4), the container hasn't exited yet, use the
+following command to forcefully shut it down:
+
+```
+docker kill [container id]
+```
+
+Note: You can also execute the Kong CLI commands inside the docker container of
+course. Execute `docker exec -it [container id] kong quit --timeout=10`. See
+[Controlled stop using the Kong CLI](#controlled-stop-using-the-kong-cli).
+
+[Back to TOC](#table-of-contents)
+
+---
+
+### Controlled stop using orchestration tools (eg. Kubernetes)
+
+For the purpose of explaining it, we'll assume Kubernetes as the orchestration
+tool, but other tools act similarly.
+
+The first problem is that Kubernetes will send a `SIGTERM` signal to the
+container to instruct it to exit. This will cause workers to forcefully exit
+and drop connections.
+
+An additional problem is that when removing a workload (Pod) from a Kubernetes
+cluster, we do not control the load balancers. So when we delete a Pod,
+Kubernetes will update it datastore and start tearing down the Pod right away.
+Because of the eventual-consistent nature of Kubernetes, chances are pretty high
+the Pod gets torn down before the load balancers, services, etc. have been
+updated. Hence this might lead to dropped connections, as Kong will not accept
+those last connections send its way (after it received the `SIGTERM` or
+`SIGQUIT` signal).
+
+Here's what we have to work with:
+
+  - the `terminationGracePeriodSeconds` property contains the Kubernetes timeout
+    after which it will forcefully tear down the container.
+  - Kubernetes will send a `SIGTERM` to the container to instruct it to exit.
+  - Before sending `SIGTERM` it will run the `PreStop` hook (synchronously, so
```

Comment:
```suggestion
  - Before sending `SIGTERM` it will run the `[PreStop](https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/)` hook (synchronously, so
```